metagpt/utils/di_graph_repository.py

```diff
@@ -19,20 +21,87 @@
 
 
 class DiGraphRepository(GraphRepository):
+    """Graph repository based on DiGraph.
+
+    This class represents a graph repository that utilizes a directed graph (DiGraph) to manage relationships
+    between entities. It inherits from the GraphRepository class, providing a common interface for graph repositories.
+
+    Attributes:
+        _repo (DiGraph): The underlying directed graph representing the repository.
+
+    Methods:
+        insert: Insert a new triple into the graph repository.
+        select: Retrieve triples from the graph repository based on specified criteria.
+        delete: Delete triples from the graph repository based on specified criteria.
+        save: Save any changes made to the graph repository.
+        name: Get the name of the graph repository.
+
+    Example:
+        di_graph_repo = DiGraphRepository(name="MyDiGraphRepo")
+        di_graph_repo.insert(subject="Node1", predicate="connects_to", object_="Node2")
+        # Represents a directed relationship: Node1 connects_to Node2
+
+    Note:
+        This class extends the GraphRepository class and is specifically designed for managing directed relationships
+        using a DiGraph.
+
+    """
+
     def __init__(self, name: str, **kwargs):
         super().__init__(name=name, **kwargs)
         self._repo = networkx.DiGraph()
 
     async def insert(self, subject: str, predicate: str, object_: str):
-        self._repo.add_edge(subject, object_, predicate=predicate)
+        """Insert a new triple into the directed graph repository.
+
+        This method adds a new triple to the underlying directed graph. The triple consists of a subject, a predicate
+        describing the relationship, and an object.
 
-    async def upsert(self, subject: str, predicate: str, object_: str):
-        pass
+        Args:
+            subject (str): The subject of the triple.
+            predicate (str): The predicate describing the relationship.
+            object_ (str): The object of the triple.
 
-    async def update(self, subject: str, predicate: str, object_: str):
-        pass
+        Returns:
+            None
+
+        Raises:
+            SomeException: Describe any exceptions that might be raised during the insertion process.
+
+        Example:
+            await my_di_graph_repo.insert(subject="Node1", predicate="connects_to", object_="Node2")
+            # Adds a directed relationship: Node1 connects_to Node2
+
+        Note:
+            Implementations should handle the insertion of triples into the directed graph.
+
+        """
```

Comment:
Most of the comments here are meaningless, please delete them and check other sections for similar issues